### PR TITLE
[bitnami/aspnet-core] Add tests and publish using VIB

### PR DESCRIPTION
**Pull request by joancafom**
### Description of the change

The main objective of this PR is to publish our Bitnami ASPNET-Core Helm Chart using VMware Image Builder. In order to do that, several changes are included:

- Adding the asset to the publishing workflow
- Increasing the existing test coverage of the asset by adding Cypress tests.
- Increasing the existing test coverage of the asset by adding GOSS tests.

### Benefits

- Ensuring higher quality of the Helm charts.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could show to be potentially flaky. 

### Applicable issues

NA

### Additional information

Tested in https://github.com/joancafom/charts/pull/67

### Checklist

- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: .vib/aspnet-core/vib-publish.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
           "url": "{SHA_ARCHIVE}",
           "path": "/bitnami/aspnet-core"
         },
-        "runtime_parameters": "InNlcnZpY2UiOgogICJwb3J0IjogODAKICAidHlwZSI6ICJMb2FkQmFsYW5jZXIi",
+        "runtime_parameters": "YXBwRnJvbUV4dGVybmFsUmVwbzoKICBlbmFibGVkOiB0cnVlCiAgY2xvbmU6CiAgICByZXBvc2l0b3J5OiAiaHR0cHM6Ly9naXRodWIuY29tL2RvdG5ldC9Bc3BOZXRDb3JlLkRvY3MuZ2l0IgogICAgcmV2aXNpb246ICJtYWluIgogIHB1Ymxpc2g6CiAgICBzdWJGb2xkZXI6ICJhc3BuZXRjb3JlL2Z1bmRhbWVudGFscy9zZXJ2ZXJzL2tlc3RyZWwvc2FtcGxlcy82LngvS2VzdHJlbFNhbXBsZSIKICBzdGFydENvbW1hbmQ6IFsiZG90bmV0IiwgIktlc3RyZWxTYW1wbGUuZGxsIl0Kc2VydmljZToKICB0eXBlOiBMb2FkQmFsYW5jZXIKICBwb3J0czoKICAgIGh0dHA6IDgwCg==",
```
Comment (FraPazGal) [inline .vib/aspnet-core/vib-publish.json:25]: Given that we are checking for the automounted folder containing the serviceAccount, I think we should start adding those values to the `runtime_parameters`

File: .vib/aspnet-core/goss/goss.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    filetype: directory
+    mode: "0777"
+    owner: root
+    group: root
+  /var/run/secrets/kubernetes.io/serviceaccount:
+    exists: true
+    filetype: directory
+    mode: "1777"
+command:
+  dotnet-cli:
```
Comment (FraPazGal) [inline .vib/aspnet-core/goss/goss.yaml:13]: I don't think this test adds any value to the chart verification. I would say checking this binary exists something to be tested in the container's test suite, wdyt?

**Review by FraPazGal — COMMENTED** [review]
Hi @joancafom! I just left two comments, could you please review them?

**Review by agomezmoron — DISMISSED** [review]
LGTM with @FraPazGal's suggestions

File: .vib/aspnet-core/goss/goss.yaml (hunk 2 above)
Comment (joancafom) [reply .vib/aspnet-core/goss/goss.yaml:13]: Agreed, thx. Updating it now.

**Review by agomezmoron — APPROVED** [review]

File: .vib/aspnet-core/vib-publish.json (hunk 1 above)
Comment (joancafom) [reply .vib/aspnet-core/vib-publish.json]: Added, thanks!

File: .vib/aspnet-core/vib-publish.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
           "url": "{SHA_ARCHIVE}",
           "path": "/bitnami/aspnet-core"
         },
-        "runtime_parameters": "InNlcnZpY2UiOgogICJwb3J0IjogODAKICAidHlwZSI6ICJMb2FkQmFsYW5jZXIi",
+        "runtime_parameters": "YXBwRnJvbUV4dGVybmFsUmVwbzoKICBlbmFibGVkOiB0cnVlCiAgY2xvbmU6CiAgICByZXBvc2l0b3J5OiAiaHR0cHM6Ly9naXRodWIuY29tL2RvdG5ldC9Bc3BOZXRDb3JlLkRvY3MuZ2l0IgogICAgcmV2aXNpb246ICJtYWluIgogIHB1Ymxpc2g6CiAgICBzdWJGb2xkZXI6ICJhc3BuZXRjb3JlL2Z1bmRhbWVudGFscy9zZXJ2ZXJzL2tlc3RyZWwvc2FtcGxlcy82LngvS2VzdHJlbFNhbXBsZSIKICBzdGFydENvbW1hbmQ6IFsiZG90bmV0IiwgIktlc3RyZWxTYW1wbGUuZGxsIl0KY29udGFpbmVyUG9ydHM6CiAgaHR0cDogODA4MApzZXJ2aWNlQWNjb3VudDoKICBjcmVhdGU6IHRydWUKICBhdXRvbW91bnRTZXJ2aWNlQWNjb3VudFRva2VuOiB0cnVlCnNlcnZpY2U6CiAgdHlwZTogTG9hZEJhbGFuY2VyCiAgcG9ydHM6CiAgICBodHRwOiA4MAo=",
```
[the remark this answers is not in the document]
Comment (joancafom) [reply .vib/aspnet-core/vib-publish.json:25]: Yes, the sample app is configured to listen to that port by default. However, I don't think we should remove this.

The `runtime_parameters` contain what sample app to use, its specific start command, and hence it does also make sense to me that they should indicate whatever other configuration it needs to work.

[the remark this answers is not in the document]
Comment (FraPazGal) [reply .vib/aspnet-core/vib-publish.json:25]: I understand what you mean here. In that case, it's fine by me. Approved 👍

**Review by FraPazGal — DISMISSED** [review]

**Review by agomezmoron — APPROVED** [review]

**Pull request merged**